### DISCUSSION TITLE
scvelo for labeling data?

### DISCUSSION THREAD

**Discussion by wgao688** [theislab/scvelo#1180] [General]
Hi @WeilerP ,
I see the citation for metabolic labeling data for Weiler et al. and see the API (https://scvelo.readthedocs.io/en/stable/api.html) listing inference for metabolic labeling data, but I do not have access to those commands when I run  scvelo 0.3.1 (python 3.10.13). Are the commands publicly available yet? Thanks!

**Comment by WeilerP** [comment]
Hi @wgao688, which part is not working? Everything works fine for me. A reproducible code snippet with a stack trace would be helpful.

**Reply by wgao688** [reply]
Hi Philipp, thanks for your response. 

I am using `scvelo v0.3.1` and have no issues following the tutorial (I know the dataset isn't a labeling experiment but wanted to start with that to confirm other functions are working on my end):

Installed via: `pip install git+https://github.com/theislab/scvelo@main`

`import scvelo as scv`
`adata = scv.datasets.pancreas()`
`scv.pp.filter_and_normalize(adata)`
`scv.pp.moments(adata)`
`scv.tl.velocity(adata, mode='stochastic')`
`scv.tl.velocity_graph(adata)`
`scv.pl.velocity_embedding(adata, basis='umap')`
`scv.pl.velocity_embedding_grid(adata, basis='umap')`
`scv.pl.velocity_embedding_stream(adata, basis='umap')`

This runs perfectly fine. However, if I try to run a command a like `scv.inference.<command>` (https://scvelo.readthedocs.io/en/stable/api.html#inference-from-metabolic-labeling-information), the option does not exist. I can run all other commands like `scv.tl.` or `scv.pl.`, etc. So it's not that I get an error code, just that I don't see the functions available to run.

Am I using the wrong version of scvelo? Do you have a vignette available for labeling data? Thank you!

**Reply by WeilerP** [reply]
Sorry, what do you mean by _"I don't see the functions available to run"_? What is the exact traceback error when you try to import the functions? Can you import them?

The analysis done for the CellRank 2 preprint is shown [here](https://theislab.github.io/cellrank2_reproducibility/labeling_kernel/inference.html); the functions importable from `scvelo.inference` are set [here](https://github.com/theislab/scvelo/blob/main/scvelo/inference/__init__.py).

**Reply by wgao688** [reply]
Hi Philipp, thanks for your help. From the links you provided, I see that I did not run this block after `import scvelo as scv`

```
from scvelo.inference import (
    get_labeling_time_mask,
    get_labeling_times,
    get_n_neighbors,
    get_obs_dist_argsort,
    get_parameters,
)
```

After that, I'm able to access to the functions (`get_labeling_time_mask`, etc.)
